shell/browser/api/electron_api_crash_reporter.cc

```diff
@@ -81,6 +85,40 @@ bool IsCrashReporterEnabled() {
 const std::map<std::string, std::string>& GetGlobalCrashKeys() {
   return GetGlobalCrashKeysMutable();
 }
+
+base::FilePath GetClientIdPath() {
+  base::FilePath path;
+  base::PathService::Get(electron::DIR_CRASH_DUMPS, &path);
+  return path.Append("client_id");
+}
+
+std::string ReadClientId() {
+  base::ThreadRestrictions::ScopedAllowIO allow_io;
+  std::string client_id;
+  // "xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx".length == 36
+  if (!base::ReadFileToStringWithMaxSize(GetClientIdPath(), &client_id, 36) ||
+      client_id.size() != 36)
+    return std::string();
+  return client_id;
```

Comment:
Do we have to check if it is a valid UUID or will breakpad take any 36 char string?

Reply:
i don't think breakpad cares at all what this string is